sdk/core/azure-json/src/main/java/com/azure/json/package-info.java

```diff
@@ -2,6 +2,499 @@
 // Licensed under the MIT License.
 
 /**
- * Contains general classes for handling JSON.
+ * <p>The Azure JSON library provides interfaces for stream-style JSON reading and writing. Stream-style reading and
+ * writing has the type itself defines how to read JSON to create and instance of itself and how it writes JSON.
+ * Azure JSON also provides the ability to provide implementations for JSON reading and writing by offering a service
+ * provider interface to load implementations from the classpath, the library provides a default implementation if
+ * one cannot be found on the classpath.</p>
+ *
+ * <h2>Getting Started</h2>
+ *
+ * <p>{@link com.azure.json.JsonSerializable} is the base of Azure JSON, it is the interface that types implement to
+ * provide stream-style JSON reading and writing functionality. The interface has a single implementable method
+ * {@link com.azure.json.JsonSerializable#toJson(com.azure.json.JsonWriter)} that defines how the object will be written
+ * as JSON, to the {@link com.azure.json.JsonWriter}, and an over-writable static method
+ * {@link com.azure.json.JsonSerializable#fromJson(com.azure.json.JsonReader)} that defines how to read an instance of
+ * the object from JSON, being read from the {@link com.azure.json.JsonReader}. The default implementation of
+ * {@link com.azure.json.JsonSerializable#fromJson(com.azure.json.JsonReader)} will throw an
+ * {@link java.lang.UnsupportedOperationException} if the static method isn't overwritten. Given that the type itself
+ * manages JSON serialization the type can be fluent, immutable, or a mix of fluent and immutable, it does not matter
+ * as all logic is self-encapsulated.</p>
+ *
+ * <p><strong>Sample: Fluent JsonSerializable</strong></p>
+ *
+ * <!-- src_embed com.azure.json.JsonSerializable.fluent -->
+ * <pre>
+ *
+ * &#47;**
+ *  * Implementation of JsonSerializable where all properties are fluently set.
+ *  *&#47;
+ * public class FluentJsonSerializableExample implements JsonSerializable&lt;FluentJsonSerializableExample&gt; &#123;
```

Comment:
To not confuse the reader, it is perhaps a good idea to simplify this class name to make it clear it is the user model type. Call it Widget or Foo or something else perhaps?